sound/soc/soc-pcm.c

```diff
@@ -2908,20 +2908,41 @@ int soc_new_pcm(struct snd_soc_pcm_runtime *rtd, int num)
 	struct snd_pcm *pcm;
 	char new_name[64];
 	int ret = 0, playback = 0, capture = 0;
+	int stream;
 	int i;
 
+	if (rtd->dai_link->dynamic && rtd->num_cpus > 1) {
+		dev_err(rtd->dev,
+			"DPCM doesn't support Multi CPU for Front-Ends yet\n");
+		return -EINVAL;
+	}
+
 	if (rtd->dai_link->dynamic || rtd->dai_link->no_pcm) {
-		cpu_dai = asoc_rtd_to_cpu(rtd, 0);
-		if (rtd->num_cpus > 1) {
-			dev_err(rtd->dev,
-				"DPCM doesn't support Multi CPU yet\n");
-			return -EINVAL;
-		}
+		for_each_rtd_cpu_dais(rtd, i, cpu_dai) {
+			cpu_dai = asoc_rtd_to_cpu(rtd, i);
+
+			if (rtd->dai_link->dpcm_playback) {
+				stream = SNDRV_PCM_STREAM_PLAYBACK;
+				if (snd_soc_dai_stream_valid(cpu_dai, stream)) {
+					playback = 1;
+				} else if (playback != 0) {
+					dev_err(rtd->card->dev,
+						"CPU Dai %d does not support playback\n", i);
```

Comment:
I prefer to print CPU DAI name which is more indicative.

Reply:
good idea, will fix as well.